fenzo-core/src/test/java/com/netflix/fenzo/plugins/WeightedAverageFitnessCalculatorTest.java

```diff
@@ -0,0 +1,46 @@
+package com.netflix.fenzo.plugins;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.netflix.fenzo.TaskRequest;
+import com.netflix.fenzo.TaskTrackerState;
+import com.netflix.fenzo.VMTaskFitnessCalculator;
+import com.netflix.fenzo.VirtualMachineCurrentState;
+import com.netflix.fenzo.plugins.WeightedAverageFitnessCalculator.WeightedFitnessCalculator;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class WeightedAverageFitnessCalculatorTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWeightsMustEqualOne() throws Exception {
+        VMTaskFitnessCalculator fitnessCalculator = mock(VMTaskFitnessCalculator.class);
+        List<WeightedFitnessCalculator> weightedFitnessCalculators = Arrays.asList(
+                new WeightedFitnessCalculator(fitnessCalculator, 0.3),
+                new WeightedFitnessCalculator(fitnessCalculator, 0.75)
+        );
+        new WeightedAverageFitnessCalculator(weightedFitnessCalculators);
+    }
+
+    @Test
+    public void testCalculatorWeightedAverage() throws Exception {
+        VMTaskFitnessCalculator fitnessCalculator1 = mock(VMTaskFitnessCalculator.class);
+        when(fitnessCalculator1.calculateFitness(any(), any(), any())).thenReturn(0.0);
```

Comment:
This return of 0.0 should result in the overall fitness value to be 0.0. See my review comments for WeightedAverageFitnessCalculator.java line 56.